stl/inc/flat_map

```diff
@@ -216,8 +216,8 @@ public:
 
     static_assert(_STD input_iterator<iterator>);
 
-    using value_compare = typename _Flat_value_compare<key_type, mapped_type, key_compare>::type;
-    using containers    = typename _Flat_Container<key_container_type, mapped_container_type>::type;
+    using value_compare = typename _Flat_value_compare<key_type, mapped_type, key_compare>::value_compare;
+    using containers    = typename _Flat_Container<key_container_type, mapped_container_type>::container;

```

Comment:
By the way, as a followup to #3894, we need to remove these `typename`s finally.